docs/RFCS/monitoring_queries_and_sessions.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: Monitoring Queries and Sessions
+- Status: in-progress
+- Start Date: 2017-05-04
+- Authors: Bilal Akhtar
+- RFC PR: https://github.com/cockroachdb/cockroach/pull/15761
+- Cockroach Issue: [#7003](https://github.com/cockroachdb/cockroach/issues/7003)
+
+# Summary
+
+This feature would add a mechanism to list currently active sessions and currently executing queries
+in the admin UI, along with their start timestamps  / durations
+
+# Motivation
+
+Currently, there's no visibility into what queries/sessions are running on the cluster at any
+given point of time. Adding visibility into this has been a common customer request for
+a while, along with the ability to terminate long-running queries and specific sessions.
+
+# Detailed design
+
+The inspiration behind some of the design comes from
+[this PR from Raphael](https://github.com/cockroachdb/cockroach/pull/10317) which implemented
+node-local session registries exposed via a virtual table.
+
+## New unique UUID identifiers for sessions and queries
+
+Currently we don't have any unique identifiers for sessions or queries. Having UUID strings
+for this purpose should guarantee uniqueness. Later on, when cancellation is implemented,
+the user would only have to specify a node ID and session/query ID in the RPC call
+for cancelling that query or session.
+
+## Node-local session and query registries
+
+The session registry will be an in-memory data structure that stores a copy of a subset
+of fields of every session on that node; particularly, only those fields that
+need to be exposed through virtual tables/RPC. Addition and deletion of new
+sessions to this slice would require acquiring a mutex lock, but none of these
+fields should ever be updated during the lifetime of a session - so session-specific
+locks are unnecessary.
+
+These fields would be stored in the shadow session object:
+- Session ID (defined earlier)
+- User
+- Start timestamp
+- Any contexts/object references useful for cancellation
```

Comment:
I left it out of the shadow object completely, but if it's really useful, we can add it back. Updating it would involve acquiring a mutex on the shadow session object.

Adding it to the unanswered questions list.